app/src/main/java/com/example/sharewhatyoucanproject/login/LoginActivity.kt

```diff
@@ -0,0 +1,87 @@
+package com.example.sharewhatyoucanproject.login
+
+import android.content.Intent
+import android.os.Bundle
+import android.provider.Settings
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.DonorDrawerActivity
+import com.example.sharewhatyoucanproject.R
+import com.example.sharewhatyoucanproject.SelectUserActivity
+import com.example.sharewhatyoucanproject.databinding.ActivityLoginBinding
+import com.example.sharewhatyoucanproject.models.UserType
+import com.example.sharewhatyoucanproject.utils.getUserType
+import com.example.sharewhatyoucanproject.utils.showToast
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    lateinit var loginViewModel: LoginViewModel
+    lateinit var auth: FirebaseAuth
+    lateinit var db: FirebaseFirestore
+    lateinit var circularProgressIndicator: com.google.android.material.progressindicator.CircularProgressIndicator
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        circularProgressIndicator = findViewById(R.id.progress_circular)
+        auth = FirebaseAuth.getInstance()
+        db = FirebaseFirestore.getInstance()
+        supportActionBar?.hide()
+        loginViewModel = ViewModelProvider(
+            this,
+            LoginViewModelFactory(this.application, auth, db),
```

Comment:
you can use default parameters in `LoginViewModelFactory` instead of passing dependencies from the activity